### DISCUSSION TITLE
[Feature] User impersonation as server admin

### DISCUSSION THREAD

**Discussion by MexHigh** [immich-app/immich#10543] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I'm the server admin for an immich instance for my whole family. I have migrated the images from Google Photos with [immich-go](https://github.com/simulot/immich-go) for most of them. I would have liked to have the ability to **log in as another user without having to know their password ("impersonation")** so I can check that immich-go has uploaded everything correctly and possibly rearrange albums and share them with the right people again before giving them access to their new account.

In my opinion, there are other use cases, such as the manual deduplication of videos (to save storage space) or the naming of people from facial recognition.

Of course, this should only be a feature for administrators and should at best be secured with additional authentication, as it is certainly easy to misuse.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Most oauth servers support this. We deliberately keep the built-in auth system as simple as possible.
